R/makeTreeSummarizedExperimentFromqiime2.R

```diff
@@ -1,93 +1,99 @@
+#'   taxonomyTableFile = taxonomyTableFile,
+#'   sampleMetaFile = sampleMetaFile,
+#'   repSeqFile = repSeqFile,
+#'   phyTreeFile = phyTreeFile
 #' )
 #'
 #' tse
-makeTreeSummarizedExperimentFromqiime2 <- function(otu_qza,
-                                                   taxa_qza = NULL,
-                                                   sam_tab = NULL,
-                                                   refseq_qza = NULL,
-                                                   tree_qza = NULL) {
-    feature_tab <- read_qza(otu_qza)
-
-    if (!is.null(taxa_qza)) {
-        taxa_tab <- read_qza(taxa_qza)
-        taxa_tab <- subset_taxa_in_feature(taxa_tab, feature_tab)
-        taxa_tab <- parse_q2taxonomy(taxa_tab)
+makeTreeSummarizedExperimentFromqiime2 <- function(featureTableFile,
+                                                   taxonomyTableFile = NULL,
+                                                   sampleMetaFile = NULL,
+                                                   featureNamesAsRepseq = TRUE,
+                                                   repSeqFile = NULL,
+                                                   phyTreeFile = NULL) {
+    feature_tab <- .read_qza(featureTableFile)
+
+    if (!is.null(taxonomyTableFile)) {
+        taxa_tab <- .read_qza(taxonomyTableFile)
+        taxa_tab <- .subset_taxa_in_feature(taxa_tab, feature_tab)
+        taxa_tab <- .parse_q2taxonomy(taxa_tab)
     } else {
-        taxa_tab <- NULL
+        taxa_tab <- S4Vectors:::make_zero_col_DataFrame(nrow(feature_tab))
     }
 
-    if (!is.null(sam_tab)) {
-        sam_tab <- read_q2sample_meta(sam_tab)
+    if (!is.null(sampleMetaFile)) {
+        sample_meta <- .read_q2sample_meta(sampleMetaFile)
     } else {
-        sam_tab <- NULL
+        sample_meta <- S4Vectors:::make_zero_col_DataFrame(ncol(feature_tab))
     }
 
-    if (!is.null(tree_qza)) {
-        tree <- read_qza(tree_qza)
+    if (!is.null(phyTreeFile)) {
+        tree <- .read_qza(phyTreeFile)
     } else {
         tree <- NULL
     }
 
     # check the row.names of feature table: DNA sequence or not
     # if is DNA, row.names(feature_tab) is set as refseq
-    if  (is_dna_seq(row.names(feature_tab))) {
+    if  (.is_dna_seq(row.names(feature_tab))) {
         refseq <- row.names(feature_tab)
         refseq_nm <- paste0("OTU", seq_along(refseq))
         names(refseq) <- refseq_nm
-
-        # set the rownames of feature and taxa tab as OTU1, OTU2,...
-        if (!is.null(taxa_tab)) {
-            rownames(taxa_tab) <- refseq_nm
-        }
-        rownames(feature_tab) <- refseq_nm
     } else {
         refseq <- NULL
     }
 
-    if (!is.null(refseq_qza)){
-        refseq <- read_qza(refseq_qza)
+    if (!is.null(repSeqFile)){
+        if (!is.null(refseq)) {
+            warning(
+                "use sequences from `repSeqFile` as representative sequences",
+                "argument `featureNamesAsRepseq` does not work"
+            )
```

Comment:
If you explain the behavior in the man page, the warning is not necessary. I would like the user to know about the behavior of a function, without having to resort to a test run.

edit:
This if can be merged, with the one before it, like this:

```
if (!is.null(repSeqFile)){
    refseq <- .read_qza(repSeqFile)
} else if(!is.null(feature_tab)) {
    refseq <- .rownames_as_dna_seq(rownames(feature_tab))
}
if(is(refseq,"DNAStringSet") && !is.null(feature_tab)) {
    refseq <- rownames(feature_tab)
    names(refseq) <- paste0("seq_", seq_along(refseq))
} 
.rownames_as_dna_seq<- function(seq){
    seq <- try({DNAStringSet(seq)}, silent = TRUE)
    if(inherits(seq, "try-error")){
        return(NULL)
    }
    seq
}
```